### PR TITLE
[cpu] remove vec code for ops that do not support complex no

### PR DESCRIPTION
**Pull request by khushi-411**
Removes dead code pertaining to ATen ops for which complex dtype is unsupported.

Reference: https://github.com/pytorch/pytorch/pull/110217#discussion_r1340599702

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10

### CODE REVIEW

**Review by sanchitintel — COMMENTED** [review]
Thanks for this PR, @khushi-411!
It looks good to me.

@vfdev-5, since you're a code-owner for the files being modified, would you be able to help review this PR? Some other code-owners may be unavailable this week. This PR removes some dead code. Thanks!

File: aten/src/ATen/cpu/vec/vec256/vec256_complex_double.h [hunk 1]
```diff
@@ -318,33 +297,9 @@ template <> class Vectorized<c10::complex<double>> {
   Vectorized<c10::complex<double>> operator!=(const Vectorized<c10::complex<double>>& other) const {
     return _mm256_cmp_pd(values, other.values, _CMP_NEQ_UQ);
   }
-  Vectorized<c10::complex<double>> operator<(const Vectorized<c10::complex<double>>&) const {
```
Comment (vfdev-5) [inline aten/src/ATen/cpu/vec/vec256/vec256_complex_double.h:321]: I wonder what would happen if user calls such operators on complex vec data? On nightly it should raise an error and when we remove it what would be the behaviour?

Comment (sanchitintel) [reply aten/src/ATen/cpu/vec/vec256/vec256_complex_double.h:321]: Oh, good point! 

@khushi-411, please only remove dead-code for ATen ops, as we rely upon [Dispatcher to raise errors for unsupported dtypes (https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/Dispatch.h#L222-L228).

Comment (khushi-411) [reply aten/src/ATen/cpu/vec/vec256/vec256_complex_double.h]: Makes sense. Thanks, @vfdev-5 and @sanchitintel!

Updated in https://github.com/pytorch/pytorch/pull/110280/commits/96987ac08a360bcadd97a40ec19ac23408e59804

**Review by vfdev-5 — APPROVED** [review]
LGTM, thanks @khushi-411

**Pull request closed without merging**